### ISSUE TITLE
[Display only] Mismatch between a Pull Request Number in commit view 

### ISSUE THREAD

**Issue by 99rgosse** [go-gitea/gitea#22525]
### Description

Hello
I noticed there is a mismatch in the commit view in all my repositories/organizations :

If you click on a commit, if it has been merged in a Pull Request, it shows on top left the number of the PR:
![image](https://user-images.githubusercontent.com/61579380/213399645-13af59c2-654f-461f-a2df-f61f2593abe4.png)

In my case, this number is always wrong : 
 it should display 2934 so it is higher than the correct one by one unit if the pull request 2935 is the next one.

I have another case where it should be 2907 but display 2909 (the 2908 PR has not yet been merged)

The Pull Request view has no problem, it displays always the correct commits (see below screenshot)

### Gitea Version

1.18.1

### Can you reproduce the bug on the Gitea demo site?

No

### Screenshots

![inconsistency](https://user-images.githubusercontent.com/61579380/213399263-77c1628b-b8e3-419c-820e-4207bc82f05f.png)

### How are you running Gitea?

Docker

### Database

PostgreSQL

**Comment by lunny** [comment]
Are their commit id the same?

**Comment by 99rgosse** [comment]
No, they don't have same commit ID, even if I checked the branch, it is the correct one from PR side
![2023_01_19_14_34_15_Window](https://user-images.githubusercontent.com/61579380/213457479-4f9f8daf-7ba1-48d2-9668-82408a55e28b.png)

**Comment by lunny** [comment]
This should be fixed by #25180
